quinn-proto/src/config.rs

```diff
@@ -836,6 +873,45 @@ impl ServerConfig {
         self.preferred_address_v6 = address;
         self
     }
+
+    /// Maximum number of [`Incoming`][crate::Incoming] to allow to exist at a time before
+    /// rejecting new [`Incoming`][crate::Incoming] automatically.
+    ///
+    /// Should hover around zero unless connection-creating packets are being received faster than
+    /// the application is calling [`Endpoint::accept`][crate::Endpoint::accept] and dealing with
+    /// them.
+    pub fn max_incoming(&mut self, max_incoming: usize) -> &mut Self {
+        self.max_incoming = max_incoming;
+        self
+    }
+
+    /// Maximum number of received bytes to buffer for each [`Incoming`][crate::Incoming] between
+    /// when the [`Incoming`][crate::Incoming] is created and when it is used to
+    /// accept/refuse/retry/ignore the connection attempt (not including the first packet).
+    ///
+    /// Packets received in excess of this limit are dropped. This may cause 0-RTT data to have to
+    /// be retransmitted, and, in some cases, failure of the handshake to complete.
```

Comment:
I don't see why this should cause connection failure except in the same sense that failing to handle an `Incoming` in a timely fashion would, which probably doesn't bear repeating here.

Reply:
You mentioned before the possibility of receiving Initial packets after the first initial packet, rather than solely just 0-RTT packets. My understanding isn't entirely clear on how that works and whether the loss of those packets could lead to handshake failure. I probably should've highlighted it more as something I was uncertain about. I removed that claim from the docs for now.

Reply:
In short, everything important is subject to retransmission until the connection times out completely. A lot of thought was put into the handshake process to avoid situations where transient packet loss could cause a deadlock.